src/core/thread/router_table.cpp

```diff
@@ -502,23 +502,25 @@ uint8_t RouterTable::GetLinkCost(Router &aRouter)
     return rval;
 }
 
-void RouterTable::ProcessTlv(const Mle::RouteTlv &aTlv)
+void RouterTable::ProcessRouterIdSet(uint8_t aRouterIdSequence, const Mle::RouterIdSet &aRouterIdSet)
 {
     bool allocationChanged = false;
 
-    mRouterIdSequence            = aTlv.GetRouterIdSequence();
+    mRouterIdSequence            = aRouterIdSequence;
     mRouterIdSequenceLastUpdated = TimerMilli::GetNow();
 
+    VerifyOrExit(mAllocatedRouterIds != aRouterIdSet);
+
     for (uint8_t routerId = 0; routerId <= Mle::kMaxRouterId; routerId++)
     {
-        if (aTlv.IsRouterIdSet(routerId) == IsAllocated(routerId))
+        if (aRouterIdSet.Contains(routerId) == IsAllocated(routerId))
         {
             continue;
         }
 
         allocationChanged = true;
 
```

Comment:
Can we consider removing the `allocationChanged` variable (and the related `if` below), since we know after the newly added `VerifyOrExit(mAllocatedRouterIds != aRouterIdSet)` check, some router id has definitely changed.

Reply:
yes, you are right!  updated.